Now that you've deployed your endpoint - it's time to slam it!

In [1]:
import os
import dotenv
dotenv.load_dotenv()
#os.environ["TOGETHER_API_KEY"] = getpass.getpass("Enter your Together API key: ")

True

Let's try with 1 request, just to verify our endpoint is alive.

Make sure you provide your own endpoint identifier! It will look something like this:

- `your-username-here/openai/gpt-oss-20b-unique-identifier`

In [ ]:
from together import Together

client = Together()

# REPLACE WITH YOUR OWN ENDPOINT IDENTIFIER
model_endpoint = "openai/gpt-oss-20b"

response = client.chat.completions.create(
    model=model_endpoint,
    messages=[
      {
        "role": "user",
        "content": "How much wood could a wood chuck chuck if a wood chuck could chuck wood?"
      }
    ]
)
print(response.choices[0].message.content)


**Short answer:**  
“About 700 pounds of wood—if a woodchuck could actually chuck wood.”

---

### Why 700 pounds?

The tongue‑twister “How much wood would a woodchuck chuck?” is mostly a play on words, but it has inspired a little bit of science. In 1988, wildlife biologist Richard Thomas ran a rough calculation:

| Item | Weight per unit | Units in a burrow | Total weight |
|------|-----------------|-------------------|--------------|
| Woodchuck body (average 5 lb) | 5 lb | 1 | 5 lb |
| Soil displaced (average 6 lb/ft³) | 6 lb | 15 ft³ (a typical burrow) | ~90 lb |
| Wood moved (assuming a 1‑ft³ volume, ~35 lb/ft³) | 35 lb | 5 ft³ (the amount a chuck might move in a day) | ~175 lb |

Thomas then multiplied the daily volume by an estimate of how many hours a woodchuck could “chuck” in a day, arriving at roughly **700 pounds (≈320 kg)** of wood over the course of a year.

> **Note:** The calculation is highly speculative—woodchucks (aka groundhogs) are burrowing animals that move dirt, not logs. The figure is more a fun illustration than a hard science fact.

---

### A more playful take

If we go purely with the rhyme:

> “A woodchuck would chuck as much wood as a woodchuck would chuck if a woodchuck could chuck wood.”

…and if we want to be even more whimsical:

> “A woodchuck could chuck all the wood it could fit in its burrow, minus the small woodchuck-sized gaps—so roughly the size of a decent chicken‑pox bed, or 700 pounds of wood for a day‑long chuck‑fest.”

---

### Bottom line

- **Real life:** Woodchucks move dirt, not wood.  
- **Tongue‑twister:** A playful rhyme with no definitive answer.  
- **Scientific estimate:** About **700 pounds** of wood in a year (according to a fun, rough calculation).

So, if a woodchuck *could* chuck wood, it would probably manage a hefty haul—just not as much as the legend of its burrowing prowess!

Now, let's SLAM IT.

In [3]:
import asyncio
from together import AsyncTogether

async def send_request(client, idx):
    try:
        response = await client.chat.completions.create(
            model=model_endpoint,
            messages=[
                {
                    "role": "user",
                    "content": f"How much wood could a wood chuck chuck if a wood chuck could chuck wood? (Request {idx})"
                }
            ]
        )
        print(f"Response {idx}: {response.choices[0].message.content[:60]}...")
    except Exception as e:
        print(f"Request {idx} failed: {e}")

async def main():
    client = AsyncTogether()
    tasks = [send_request(client, i,) for i in range(1, 25)]
    await asyncio.gather(*tasks)

# Run the async main function
await main()

Response 1: **The classic answer**

> “A woodchuck would chuck as much w...
Response 21: **Answer (Tongue‑twister style)**  
A woodchuck would chuck ...
Response 15: **Answer:**  
*A wood‑chuck would chuck as much wood as a wo...
Response 19: **Answer (Request 19)**  

The old tongue‑twister is a bit o...
Response 6: Ah, the classic tongue‑twister that has baffled poets, prank...
Response 17: If a wood‑chuck could chuck wood, it would probably lug **ab...
Response 13: **Answer (classic tongue‑twister style):**

> *“A wood‑chuck...
Response 18: **Answer to the classic tongue‑twister**  
> “How much wood ...
Response 7: That classic tongue‑twister doesn’t have a hard‑and‑fast num...
Request 22 failed: Error code: 503 - The server is overloaded or not ready yet.
Request 11 failed: Error code: 503 - The server is overloaded or not ready yet.
Request 4 failed: Error code: 503 - The server is overloaded or not ready yet.
Request 9 failed: Error code: 503 - The server is overloaded or not rea

#### Output of 25 requests processed asynchronously  

Response 1: **The classic answer**   > “A woodchuck would chuck as much w...  
Response 21: **Answer (Tongue‑twister style)**    A woodchuck would chuck ...    
Response 15: **Answer:** A wood‑chuck would chuck as much wood as a wo...    
Response 19: **Answer (Request 19)** The old tongue‑twister is a bit o...  
Response 6: Ah, the classic tongue‑twister that has baffled poets, prank...  
Response 17: If a wood‑chuck could chuck wood, it would probably lug **ab...  
Response 13: **Answer (classic tongue‑twister style):**  > *“A wood‑chuck...  
Response 18: **Answer to the classic tongue‑twister**    > “How much wood ...  
Response 7: That classic tongue‑twister doesn’t have a hard‑and‑fast num...  
Request 22 failed:<mark> Error code: 503</mark>  - The server is overloaded or not ready yet.  
Request 11 failed:<mark> Error code: 503</mark>  - The server is overloaded or not ready yet.  
Request 4 failed: <mark>Error code: 503 </mark> - The server is overloaded or not ready yet.  
Request 9 failed: <mark>Error code: 503 </mark> - The server is overloaded or not ready yet.  
Response 10: Here are ten different ways to answer the age‑old tongue‑twi...  
Response 5: **Answer (Request 5)**  A classic 1988 study by wildlife bi...    
Response 23: A classic tongue‑twister! In the spirit of the joke the answ...  
Response 24: ["A woodchuck would chuck as much wood as a woodchuck could ...  
Response 20: Here are **20 whimsical, scientific, and tongue‑twister‑styl...  
Response 16: A wood chuck would chuck as much wood as a wood chuck could ...  
Response 14: A classic tongue‑twister answer is:  > “A woodchuck would ch...  
Response 8: **The tongue‑twister answer (in a nutshell):** About **700 p...  
Response 2: **Answer (Request 2)**    > “If a wood‑chuck could chuck woo...  
Response 12: **Answer to Request 12**  > *“How much wood could a wood‑chu...    
Response 3: A woodchuck would chuck roughly **700 pounds of wood per day...  

In [ ]:
messages = [
    (
        "system",
        "You are a helpful assistant that translates English to French. Translate the user sentence.",
    ),
    ("human", "I love programming."),
]
ai_msg = llm.invoke(messages)
ai_msg